src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/ConfirmServiceCallbackHandler.java

```diff
@@ -73,44 +86,64 @@ private CallbackResponse checkServedDocumentsOtherHasWhiteSpace(CallbackParams c
 
         ServiceMethod serviceMethod = caseData.getServiceMethod();
-
+        Map<String, Object> data = callbackParams.getRequest().getCaseDetails().getData();
         // TODO: this field will be different (date / date time) in CCD depending on service method.
         LocalDate serviceDate = caseData.getServiceDate();
 
         LocalDate deemedDateOfService = serviceMethod.getDeemedDateOfService(serviceDate);
         LocalDateTime responseDeadline = addFourteenDays(deemedDateOfService);
-
         data.put("deemedDateOfService", deemedDateOfService);
         data.put("responseDeadline", responseDeadline);
 
+        CaseData caseDateUpdated = caseData.toBuilder()
+            .deemedDateOfService(deemedDateOfService)
+            .responseDeadline(responseDeadline)
+            .build();
+
+        CaseDocument certificateOfService = certificateOfServiceGenerator.generate(
+            caseDateUpdated,
+            callbackParams.getParams().get(BEARER_TOKEN).toString()
+        );
+        List<Element<CaseDocument>> systemGeneratedCaseDocuments = caseData.getSystemGeneratedCaseDocuments();
+        if (ObjectUtils.isEmpty(systemGeneratedCaseDocuments)) {
+            data.put("systemGeneratedCaseDocuments", wrapElements(certificateOfService));
+        } else {
+            systemGeneratedCaseDocuments.add(element(certificateOfService));
+            data.put("systemGeneratedCaseDocuments", systemGeneratedCaseDocuments);
+        }
+
         return AboutToStartOrSubmitCallbackResponse.builder()
             .data(data)
             .build();
     }
 
     private SubmittedCallbackResponse buildConfirmation(CallbackParams callbackParams) {
-        Map<String, Object> data = callbackParams.getRequest().getCaseDetails().getData();
-        CaseData caseData = mapper.convertValue(data, CaseData.class);
+        CaseData caseData = caseDetailsConverter.toCaseData(callbackParams.getRequest().getCaseDetails());
 
         LocalDate deemedDateOfService = caseData.getDeemedDateOfService();
         String formattedDeemedDateOfService = formatLocalDate(deemedDateOfService, DATE);
         String responseDeadlineDate = formatLocalDateTime(addFourteenDays(deemedDateOfService), DATE_TIME_AT);
-        String certificateOfServiceLink = "http://www.google.com";
-
-        String body = format("<br /> Deemed date of service: %s."
-                                 + "<br />The defendant must respond before %s."
-                                 + "\n\n[Download certificate of service](%s) (PDF, 266 KB)",
-                             formattedDeemedDateOfService, responseDeadlineDate, certificateOfServiceLink
+        Long documentSize = ElementUtils.unwrapElements(caseData.getSystemGeneratedCaseDocuments()).stream()
+            .filter(c -> c.getDocumentType() == DocumentType.CERTIFICATE_OF_SERVICE)
+            .findFirst()
+            .map(CaseDocument::getDocumentSize)
+            .orElse(0L);
```

Comment:
is this right to put document size = 0 when it isn't there?

Reply:
Not sure what is the requirement in that case, following the consistent behaviour as we are doing with sealed claim. At this stage the document must be generated. This is kind of very edge case.